base/src/test/java/io/spine/base/EnvironmentTest.java

```diff
@@ -217,32 +232,63 @@ void determineUsingType() {
     @Test
     @DisplayName("detect the current custom environment in presence of custom types")
     void determineUsingTypeInPresenceOfCustom() {
-        register(new Staging(), new Local());
+        environment.register(Local.class)
+                   .register(Staging.class);
 
         assertThat(environment.is(Local.class)).isTrue();
     }
 
     @Test
     @DisplayName("return the instance of the default environment type")
     void returnInstance() {
-        assertThat(environment.type()).isInstanceOf(Tests.class);
+        assertThat(environment.type()).isSameInstanceAs(Tests.class);
     }
 
     @Test
     @DisplayName("return the instance of a custom environment type")
-    void returnCustomInstnace() {
-        register(new Local(), new Staging());
+    void returnCustomInstance() {
+        environment.register(Local.class)
+                   .register(Staging.class);
 
-        assertThat(environment.type()).isInstanceOf(Local.class);
+        assertThat(environment.type()).isSameInstanceAs(Local.class);
     }
 
-    private static void register(EnvironmentType... types) {
-        for (EnvironmentType type : types) {
-            Environment.instance()
-                       .register(type);
-        }
+    @Test
+    @DisplayName("cache a custom environment type")
+    void cacheCustom() {
+        environment.register(Staging.class);
+
+        Staging.set();
+        assertThat(environment.is(Staging.class)).isTrue();
+
+        Staging.reset();
+        assertThat(new Staging().enabled()).isFalse();
+        assertThat(environment.is(Staging.class)).isTrue();
     }
 
+    @Test
+    @DisplayName("cache the `Tests` environment type")
+    void cacheTests() throws InterruptedException {
+        AtomicBoolean envCached = new AtomicBoolean(false);
+        assertThat(environment.is(Tests.class));
+        Thread thread = new Thread(() -> {
+            /*
```

Comment:
Please convert this to the method-level comment.

Don't use inline comments except for emergency cases.